### PR TITLE
[new release] dolmen_type, dolmen_model, dolmen_lsp, dolmen_loop, dolmen_bin and dolmen (0.8.1)

### PR DESCRIPTION
**Pull request by Gbury**
A typechecker for automated deduction languages

- Project page: <a href="https://github.com/Gbury/dolmen">https://github.com/Gbury/dolmen</a>
- Documentation: <a href="https://gbury.github.io/dolmen">https://gbury.github.io/dolmen</a>

### UI

- Fix handling of size/time limits on windows (PR#117)
- Fix spurious printing of backtraces (PR#118)
- Add release binaries for windows

### LSP

- Add option for the lsp to read preludes before checking
  each file (PR#116)
- The LSP now sends an empty list of diagnostics upon closing
  a file (PR#116)

### Parsing

- Fix a bug related to alt-ergos function definition, which
  were previously alwyas non-recursive. Now, alt-ergo's function
  definitions are always recursive (PR#123)
- Add `parse_raw_lazy` to parse a string into a lazy list of
  statements (PR#125)

### Typing

- Properly add binding locations for implicit type variables
  (PR#123)
- Ensure that type of recursively defined symbols are freshened
  to avoid type variables sharing between declaration and
  definition (PR#123)

### Loop

- New module to implement Alarms (size/time limits) (PR#117)
- Add optional argument to `Pipeline.run` to specify
  an alarm implementation (PR#117)
- Add a `bt` key to the state to record whether we should print
  backtraces (PR#118)
- Use `parse_raw_lazy` to parse raw contents in full mode if/when
  necessary (PR#125)

### CODE REVIEW

File: packages/dolmen/dolmen.0.8.1/opam [hunk 1]
```diff
@@ -0,0 +1,46 @@
+opam-version: "2.0"
+maintainer: "Guillaume Bury <guillaume.bury@gmail.com>"
+authors: "Guillaume Bury <guillaume.bury@gmail.com>"
+license: "BSD-2-Clause"
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
```
Comment (raphael-proust) [inline packages/dolmen/dolmen.0.8.1/opam:7]: Any reason not to allow the installation to run the tests of the package?

```suggestion
  ["dune" "build" "-p" name "-j" jobs "@runtest" {with-test}]
```

File: packages/dolmen_bin/dolmen_bin.0.8.1/opam [hunk 2]
```diff
@@ -0,0 +1,42 @@
+opam-version: "2.0"
+maintainer: "Guillaume Bury <guillaume.bury@gmail.com>"
+authors: "Guillaume Bury <guillaume.bury@gmail.com>"
+license: "BSD-2-Clause"
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
```
Comment (raphael-proust) [inline packages/dolmen_bin/dolmen_bin.0.8.1/opam:7]: Same remark here and in other packages.

File: packages/dolmen/dolmen.0.8.1/opam (hunk 1 above)
Comment (Gbury) [reply packages/dolmen/dolmen.0.8.1/opam:7]: there's no reason, just that I wrote the file by hand some time ago, and didn't update it in a while. I'll add the test part (and probably also the doc part I suppose).

**Comment by Gbury** [comment]
I updated the build instructions to also build tests and doc when appropriate (matching commit in the dolmen repo : https://github.com/Gbury/dolmen/commit/18b02c1c23598df5c09601667093e1ff5f42ab43 )

File: packages/dolmen_bin/dolmen_bin.0.8.1/opam (hunk 2 above)
Comment (Gbury) [reply packages/dolmen_bin/dolmen_bin.0.8.1/opam]: fixed

**Comment by Gbury** [comment]
So, the addition of the tests showed that some tests that should only be run on the main package (`dolmen`) are also run on other packages (because I forgot a `package` stanza in the corresponding dune file), so as a workaround I added `mdx` as a test-only dependency on all of the `dolmen_*` packages (which is not a problem, since `dolmen` already has such a dependency, and all other packages depend on `dolmen`).
I also committed the correct fix in the dolmen repo (see https://github.com/Gbury/dolmen/commit/2cddfe2d24e45e58d05696f18e9af984fe3e41f7 )

**Comment by Gbury** [comment]
I'm investigating the 5.0 failures

**Comment by Gbury** [comment]
The 5.0 failures are related to the following bug in ocaml 5.0 : https://github.com/ocaml/ocaml/issues/11995
I have a way to fix the code of dolmen to make it compatible (and bug-free) for all versions of the compiler, so I'll redo the 0.8.1 release with the fix, and add conflicts between ocaml `5.0` and older releases of dolmen.

**Comment by Gbury** [comment]
So, I re-did the `0.8.1` release, including a fix that should make dolmen work on ocaml `5.0.0`. Hopefully, the CI should now be happy, ^^

**Comment by kit-ty-kate** [comment]
Thanks

**Pull request merged**